igor-web/src/main/groovy/com/netflix/spinnaker/igor/wercker/WerckerBuildMonitor.groovy

```diff
@@ -0,0 +1,335 @@
+            } else {
+                Map<String, List<Run>> runs = werckerService.getRunsSince(since)
+                runs.keySet().forEach( { pipeline ->
+                    processRuns(werckerService, master, pipeline, delta, runs.get(pipeline))
+                } )
+            }
+        } catch (e) {
+            log.error("Error processing runs for Wercker[{}]", kv("master", master), e)
+        }
+        log.debug("Took ${System.currentTimeMillis() - startTime}ms to retrieve Wercker pipelines (master: {})", kv("master", master))
+        return new PipelinePollingDelta(master: master, items: delta)
+    }
+
+    Set<String> getConfiguredJobs(String master) {
+        Set<String> jobs = []
+        front50.get().getAllPipelineConfigs().forEach({ pipeline ->
+            pipeline['triggers'].forEach({ trigger ->
+                if (trigger['enabled'] && trigger['type'] == 'wercker' && trigger['master'] == master) {
+                    log.debug "configured Trigger for ${master} pipeline: ${pipeline['application']} ${pipeline['name']} ${trigger}"
+                    jobs.add(trigger['job'])
+                }
+            })
+            pipeline['stages'].forEach({ stage ->
+                if (stage['type'] == 'wercker' && stage['master'] == master) {
+                    log.debug "configured Stage for ${master} pipeline: ${pipeline['application']} ${pipeline['name']} ${stage}"
+                    jobs.add(stage['job'])
+                }
+            })
+        })
+        return jobs
+    }
+
+    static Comparator<Run> finishedAtComparator = new Comparator<Run>() {
+        @Override
+        public int compare(Run r1, Run r2) {
+            if (r1.finishedAt != null && r2.finishedAt != null) {
+                return r1.finishedAt.fastTime - r2.finishedAt.fastTime
+            } else if (r1.finishedAt == null) {
+                return r2.finishedAt == null? 0 : -1
+            } else {
+                return 1
+            }
+        }
+    }
+
+    Run getLastFinishedAt(List<Run> runs) {
+        return (runs && runs.size() > 0) ? Collections.max(runs, finishedAtComparator) : null
+    }
+
+    static Comparator<Run> startedAtComparator = new Comparator<Run>() {
+        @Override
+        public int compare(Run r1, Run r2) {
+            if (r1.startedAt != null && r2.startedAt != null) {
+                return r1.startedAt.fastTime - r2.startedAt.fastTime
+            } else {
+                return (r1.startedAt == null) ? ((r2.startedAt == null)? 0 : -1) : 1
+            }
+        }
+    }
+
+    Run getLastStartedAt(List<Run> runs) {
+        return Collections.max(runs, startedAtComparator)
+    }
+
+    /**
+     * wercker.pipeline = project|job
+     * wercker.run = build
+     */
+    private void processRuns( WerckerService werckerService, String master, String pipeline,
+            List<PipelineDelta> delta, List<Run> runs) {
+        List<Run> allRuns = runs? runs : werckerService.getBuilds(pipeline)
```

Comment:
`runs ?: werkerService.getBuilds(pipeline)` is more groovy-like

Reply:
Fixed elvis